pkg/k8s/rule_translate.go

```diff
@@ -234,16 +235,25 @@ func PreprocessRules(r api.Rules, cache *ServiceCache) error {
 		}
 		for ns, ep := range cache.endpoints {
 			svc, ok := cache.services[ns]
-			if ok && svc.IsExternal() {
-				eps := ep.GetEndpoints()
-				if eps != nil {
-					t := NewK8sTranslator(ns, *eps, false, svc.Labels, false)
-					// We don't need to check the translation result here because the k8s
-					// RuleTranslator above sets allocatePrefixes to be false.
-					err := t.Translate(rule, &policy.TranslationResult{})
-					if err != nil {
-						return err
-					}
+			// Normally, only services without a label selector (i.e. empty services)
+			// are allowed as targets of a toServices rule.
+			// This is to minimize the chances of a pod IP being selected by this rule, which might
+			// cause conflicting entries in the ipcache.
+			//
+			// This requirement, however, is dropped for HighScale IPCache mode, because pod IPs are
+			// normally excluded from the ipcache regardless.
+			if !ok || (!option.Config.EnableHighScaleIPcache && !svc.IsExternal()) {
+				continue
+			}
+
+			eps := ep.GetEndpoints()
+			if eps != nil {
+				t := NewK8sTranslator(ns, *eps, false, svc.Labels, false)
```

Comment:
https://github.com/cilium/cilium/pull/25687 is nearing a merge so this will conflict.

Reply:
rebased